tests/qchem/of_tests/test_molecular_hamiltonian.py

```diff
@@ -105,6 +105,53 @@ def test_building_hamiltonian(
     assert qubits == 2 * nact_orbs
 
 
+@pytest.mark.parametrize(
+    (
+        "charge",
+        "mult",
+        "package",
+        "nact_els",
+        "nact_orbs",
+        "mapping",
+    ),
+    [
+        (0, 1, "pyscf", 2, 2, "jordan_WIGNER"),
+        (2, 1, "pyscf", 2, 2, "BRAVYI_kitaev"),
+    ],
+)
+@pytest.mark.usefixtures("skip_if_no_openfermion_support", "use_legacy_and_new_opmath")
+def test_building_hamiltonian_molecule_class(
+    charge,
+    mult,
+    package,
+    nact_els,
+    nact_orbs,
+    mapping,
+    tmpdir,
+):
+    r"""Test that the generated Hamiltonian `built_hamiltonian` using the molecule class, is an
+    instance of the PennyLane Hamiltonian class and the correctness of the total number of qubits
+    required to run the quantum simulation. The latter is tested for different values of the
+    molecule's charge and for active spaces with different size"""
+
+    args = qchem.Molecule(test_symbols, test_coordinates, charge=charge, mult=mult)
+    kwargs = {
+        "method": package,
+        "active_electrons": nact_els,
+        "active_orbitals": nact_orbs,
+        "mapping": mapping,
+        "outpath": tmpdir.strpath,
+    }
+
+    built_hamiltonian, qubits = qchem.molecular_hamiltonian(args, **kwargs)
+
+    if active_new_opmath():
+        assert not isinstance(built_hamiltonian, qml.Hamiltonian)
```

Comment:
Does this mean that `built_hamiltonian` doesn't return `qml.ops.LinearCombination` under the `active_new_opmath`?

Reply:
Just checked it doesn't. We should possibly make it do that in the future.